### ISSUE TITLE
llama2 provides inconsistent replies 

### ISSUE THREAD

**Issue by emil-jose** [h2oai/h2ogpt#727]
I am trying to run llama2 7b "python generate.py --base_model=h2oai/h2ogpt-4096-llama2-7b-chat --langchain_mode='UserData' --user_path=user_path --score_model=None --share=False --db_type=weaviate". while running this I am getting the following warning as well,
`The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.`
Even when you type "Hi", the response is entirely something weird as below
`hi, I'm a 3D artist and illustrator with over 10 years of experience in the field. My expertise includes creating detailed and realistic 3D models, textures, and animations for various industries such as video games, architectural visualization, and product design. I have a strong portfolio that showcases my skills in creating high-quality 3D artwork, and I am confident in my ability to deliver exceptional results for any project.
Some of my key skills include:

3D modeling and sculpting using software such as Blender, Maya, and ZBrush
Texturing and shading techniques to create realistic materials and environments
Animation and rigging to bring characters and objects to life
Lighting and compositing to create stunning visuals
Proficiency in industry-standard software such as Autodesk 3ds Max, Adobe Photoshop, and Substance Painter
I am highly motivated and passionate about my work, and I am always looking for new challenges and opportunities to push my skills further. I am excited to collaborate with clients and help bring their ideas to life through my 3D artwork. Please feel free to reach out to me if you have any questions or would like to discuss a project.`

**Comment by pseudotensor** [comment]
Are you running with data?

I ran your command and this is what I get:

![image](https://github.com/h2oai/h2ogpt/assets/2249614/9c66f451-3a26-4715-b7a8-487a604b621c)

The "The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's attention_mask to obtain reliable results." has no effect, just warning.

If you run with data, perhaps the model got confused by the data if the data contains a resume or something.  Smaller 7B models, even llama2, can have trouble.

If you have data and can share an example data that by itself leads to a confusing output, we can take a look.  Some better prompting may help.

**Comment by emil-jose** [comment]
Thank you for the reply. I am using data (some private docs). I was wondering whether there are any performance difference wrt type of db, chroma vs weaviate?

**Comment by pseudotensor** [comment]
Probably not significant.  I suspect main issue is whether model understands the prompting.  You can change the prompting for queries and summary in expert settings or command line options, or go further and edit the code for the template in src/gpt_langchain.py

**Comment by emil-jose** [comment]
@pseudotensor I tried using llama2 13b with the following command `python3 generate.py --base_model==h2oai/h2ogpt-4096-llama2-13b-chat --load_8bit=True  --score_model=None --langchain_mode='UserData' --user_path=user_path --share=False --db_type=weaviate`, when I use the resource as llm and just type hi I am getting the following weird reposnse
![Screenshot 2023-08-29 at 13 48 32](https://github.com/h2oai/h2ogpt/assets/39084113/f88b8f89-24dc-43b2-acfe-5f3ea87d099a)

**Comment by pseudotensor** [comment]
You have an `==` in your command.  If I run:
```
python3 generate.py --base_model=h2oai/h2ogpt-4096-llama2-13b-chat --load_8bit=True  --score_model=None --langchain_mode='UserData' --user_path=user_path --share=False --db_type=weaviate
```

I get this:

![image](https://github.com/h2oai/h2ogpt/assets/2249614/db6a13d8-c1ca-4c80-955b-93e5093ce351)


I think you must have a data collection selected.  Note that "hi" will be "similar" to some documents and return results, and models will try to infer what you mean if vague.
